### DISCUSSION TITLE
Does the current version of Rook-Ceph support Ceph Crimson?

### DISCUSSION THREAD

**Discussion by CrossainQi** [rook/rook#12901] [Q&A] [answered]
I am currently using Rook-Ceph(version 11.4) and I'm curious about the support for Ceph Crimson as a storage backend. I'd like to know if the current version of Rook supports Ceph Crimson since I cannot find key-words like **seastore** in the source code and, if so, how to configure it. If there is support for Ceph Crimson, I would appreciate detailed configuration information.

If the current version of Rook does not support Ceph Crimson, do you have any plans to support it in the future?

**Comment by parth-gr** [comment]
I think so we don't support crimson, and it still needs some time to get support from Ceph.
@travisn for confirmation

**Reply by travisn** [reply, marked as answer]
Last I tested with Crimson, Rook was able to successfully run with the custom Crimson image. Crimson can be used with bluestore, which is how I tested it. Rook doesn't yet have an option for enable seastore instead of bluestore. Certainly, we expect to support it when it's getting closer to production ready. @rzarzynski Anything to add on the readiness of Crimson/seastore for testing with Rook?

Crimson is just not included in the Ceph image, so are you building the image yourself for testing?

**Reply by rzarzynski** [reply]
@travisn: I think you're correct.
The thing I would like to point out is there recent changes in Rook to support RDR aka ESB aka `bluestore-rdr` could useful in the future for SeaStore.

**Reply by travisn** [reply]
Yes #12507 enabled to convert between OSDs, which will be very useful when SeaStore is available.
